packages/sources/blocksize-capital-test/src/endpoint/price.ts

```
@@ -0,0 +1,138 @@
+    description: 'The symbol of symbols of the currency to query',
+    required: false,
+  },
+  quote: {
+    aliases: ['to', 'market'],
+    type: 'string',
+    description: 'The symbol of the currency to convert to',
+    required: false,
+  },
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: { base: string; quote: string }
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    WsMessage: Message[]
+  }
+}
+
+type WebsocketTransportGenerics = TransportGenerics & {
+  Provider: {
+    WsMessage: unknown
+  }
+}
+
+export class BlocksizeWebsocketReverseMappingTransport<
+  T extends WebsocketTransportGenerics,
+  K,
+> extends WebsocketReverseMappingTransport<T, K> {
+  api_key = ''
+}
+
+export const websocketTransport: BlocksizeWebsocketReverseMappingTransport<EndpointTypes, string> =
+  new BlocksizeWebsocketReverseMappingTransport<EndpointTypes, string>({
+    url: ({ adapterConfig: { WS_API_ENDPOINT, API_KEY } }) => {
+      websocketTransport.api_key = API_KEY
+      return WS_API_ENDPOINT
+    },
+    handlers: {
+      open: (connection) => {
+        connection.send({
+          jsonrpc: '2.0',
+          method: 'authentication_logon',
+          params: { api_key: websocketTransport.api_key },
+        })
+      },
+      message: (message) => {
+        if (Object.keys(message).length === 0) {
+          logger.debug('WS message is empty, skipping')
+          return []
+        }
+        const [_, msg] = message
+        if (!(msg.method === 'vwap' || 'method' in msg)) return []
+        const [updates] = msg.params.updates
+        const params = websocketTransport.getReverseMapping(updates.ticker)
+        if (!params) {
+          return undefined
+        }
+        const base = params.base
+        const quote = params.quote
+        return [
+          {
+            params: { base, quote },
+            response: {
+              result: updates.price as number,
```

Instead coercing price here I think we should have one more validation before we return results to check if price is `undefined` in the message. If it is we should return an error response saying something like the data provider didn't provide results.

---

Done